src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -0,0 +1,916 @@
+        /// However, the initial dispatch always obeys the switch statement sequentual
+        /// semantics. We simulate that here.
+        /// </summary>
+        private LabelSymbol BindConstantTarget(DecisionTree decisionTree)
+        {
+            var expression = decisionTree.Expression;
+
+            switch (decisionTree.Kind)
+            {
+                case DecisionTree.DecisionKind.ByType:
+                    {
+                        var byType = (DecisionTree.ByType)decisionTree;
+                        if (expression.ConstantValue?.IsNull == true)
+                        {
+                            if (byType.WhenNull != null) return BindConstantTarget(byType.WhenNull);
+                        }
+                        if (expression.ConstantValue?.IsNull != true)
+                        {
+                            foreach (var td in byType.TypeAndDecision)
+                            {
+                                var type = td.Key;
+                                var decision = td.Value;
+                                switch (Binder.Conversions.ExpressionOfTypeMatchesPatternType(expression.Type, type))
+                                {
+                                    case null:
+                                        return null; // we don't know if this matches the input
+                                    case true:
+                                        if (expression.Type.CanBeAssignedNull() && expression.ConstantValue == null)
+                                        {
+                                            return null; // we don't know if the input is null; unknown result
+                                        }
+                                        else
+                                        {
+                                            return BindConstantTarget(decision);
+                                        }
+                                    case false:
+                                        continue;
+                                }
+                            }
+                        }
+                        return (byType.Default != null) ? BindConstantTarget(byType.Default) : null;
+                    }
+                case DecisionTree.DecisionKind.ByValue:
+                    {
+                        var byValue = (DecisionTree.ByValue)decisionTree;
+                        if (byValue.Expression.ConstantValue == null)
+                        {
+                            if (byValue.ValueAndDecision.Count != 0) return null; // can't tell which decision to take
+                        }
+                        else
+                        {
+                            foreach (var vd in byValue.ValueAndDecision)
+                            {
+                                var value = vd.Key;
+                                var decision = vd.Value;
```

Comment:
vd.Value cannot be null when we arrive here except in error situations.

---
In reply to: [64668807](https://github.com/dotnet/roslyn/pull/11485#discussion_r64668807) [](ancestors = 64668807)